staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/marshal_test.go

```diff
@@ -54,9 +59,107 @@ func TestJSONSchemaPropsOrBoolUnmarshalJSON(t *testing.T) {
 			t.Errorf("Failed to unmarshal input '%v': expected %+v, got %+v", c.input, c.result, result)
 		}
 	}
+
+	t.Run("null", func(t *testing.T) {
+		var actual JSONSchemaPropsOrBool
+		err := json.Unmarshal([]byte(`null`), &actual)
+		if err == nil {
+			t.Fatal("expected error")
+		}
+	})
```

Comment:
Does this invoke the UnmarshalJSON method? I was expecting that you would have to call `actual.UnmarshalJSON` directly (not indirectly by calling `json.Unmarshal`).